sbi/src/tsm.rs

```diff
@@ -171,6 +171,103 @@ impl TvmCpuRegister {
     }
 }
 
+/// Identifies a register set in the vCPU shared-memory state area layout.
+///
+/// Register sets are roughly grouped by the specification or extension that defines them.
+/// While it is likely that registers from multiple register sets will be used in the course
+/// of handling a vCPU exit, the register sets are grouped in this way in order to allow for
+/// extensibility by future extensions, and to avoid being overly prescriptive with regards
+/// to TSM or hypervisor implementation.
+#[repr(u16)]
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum RegisterSetId {
+    /// General purpose registers.
+    Gprs = 0,
+    /// Supervisor CSRs.
+    SupervisorCsrs = 1,
+    /// HS-level CSRs.
+    HypervisorCsrs = 2,
+}
+
+impl RegisterSetId {
+    /// Returns the `RegisterSetId` corresponding to `val`.
+    pub fn from_raw(val: u16) -> Result<Self> {
+        match val {
+            0 => Ok(RegisterSetId::Gprs),
+            1 => Ok(RegisterSetId::SupervisorCsrs),
+            2 => Ok(RegisterSetId::HypervisorCsrs),
+            _ => Err(Error::InvalidParam),
+        }
+    }
+}
+
+/// Specifies the location of a particular register set in the vCPU shared-memory state area.
+#[repr(C)]
+#[derive(Clone, Copy, Debug)]
+pub struct RegisterSetLocation {
+    /// The type of the register set, identified by the `RegisterSetId` enum.
+    pub id: u16,
+    /// The version of the register set structure.
+    ///
+    /// As the definition of a particular register set may change across revisions of this API,
+    /// this version number can be used to identify the specific layout that is being used.
+    pub version: u16,
+    /// The offset of the register set from the start of the vCPU's shared-memory state area.
+    pub offset: u32,
+}
+
+/// General purpose registers. Structure for register sets of type `RegisterSetId::Gprs`.
```

Comment:
A comment here that the TSM will only read/write the minimum subset of registers needed to complete an action(and not leak other guest data) would be helpful.

Reply:
Believe this comment can be marked as resolved.